pytradfri/device/air_purifier.py

```diff
@@ -15,22 +17,43 @@
     ATTR_AIR_PURIFIER_MODE,
     ATTR_AIR_PURIFIER_MODE_AUTO,
     ATTR_AIR_PURIFIER_MOTOR_RUNTIME_TOTAL,
-    ROOT_AIR_PURIFIER,
+    ATTR_ID,
 )
-from ..typing import AirPurifierResponse
 
 if TYPE_CHECKING:
     # avoid cyclic import at runtime.
     from . import Device
 
 
+class AirPurifierResponse(BaseModel):
+    """Represent the air purifier part of the device response."""
+
+    air_quality: int = Field(alias=ATTR_AIR_PURIFIER_AIR_QUALITY)
+    controls_locked: int = Field(alias=ATTR_AIR_PURIFIER_CONTROLS_LOCKED)
+    fan_speed: int = Field(alias=ATTR_AIR_PURIFIER_FAN_SPEED)
+    filter_lifetime_remaining: int = Field(
+        alias=ATTR_AIR_PURIFIER_FILTER_LIFETIME_REMAINING
+    )
+    filter_lifetime_total: int = Field(alias=ATTR_AIR_PURIFIER_FILTER_LIFETIME_TOTAL)
+    filter_runtime: int = Field(alias=ATTR_AIR_PURIFIER_FILTER_RUNTIME)
+    filter_status: int = Field(alias=ATTR_AIR_PURIFIER_FILTER_STATUS)
+    id: int = Field(alias=ATTR_ID)
+    leds_off: int = Field(alias=ATTR_AIR_PURIFIER_LEDS_OFF)
+    mode: int = Field(alias=ATTR_AIR_PURIFIER_MODE)
+    motor_runtime_total: int = Field(alias=ATTR_AIR_PURIFIER_MOTOR_RUNTIME_TOTAL)
+
+
 class AirPurifier:
     """Represent an air purifier."""
 
+    _model_class: type[AirPurifierResponse] = AirPurifierResponse
+    raw: AirPurifierResponse
+
     def __init__(self, device: Device, index: int) -> None:
         """Create object of class."""
         self.device = device
         self.index = index
+        self.raw = self.device.raw.air_purifier[index]
```

Comment:
We can't set `raw` as an instance attribute since the `device.raw` attribute is replaced during an update.